### PR TITLE
Charts Refactoring

### PR DESCRIPTION
**Pull request by MarcusNotheis**
**Thank you for your contribution!** 👏

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [ ] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md#contribute-code) section 
- [ ] [Correct commit message style](https://github.com/SAP/ui5-webcomponents-react/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: packages/charts/src/components/ColumnChart/index.tsx [hunk 1]
```diff
@@ -1,87 +1,66 @@
...
-      options
-    } = this.props as ColumnChartPropTypes & ChartInternalProps;
+      options,
+      width,
+      height,
+      noLegend
+    } = props;
+
+    const theme: any = useTheme();
+    const data = populateData(labels, datasets, colors, theme.theme);
```
Comment (vbersch) [inline packages/charts/src/components/ColumnChart/index.tsx]: populateData returns a new instance on every render which might cause unneccessary rerenders.
What about a useChartData hook which uses useMemo inside?

File: packages/charts/src/components/DonutChart/index.tsx [hunk 2]
```diff
@@ -1,39 +1,34 @@
...
-    } = this.props as DonutChartPropTypes & ChartInternalProps;
+      options,
+      width,
+      height,
+      noLegend
+    } = props;
 
-    const doughnut = populateData(labels, datasets, colors, theme.theme, true);
+    const theme: any = useTheme();
+    const data = populateData(labels, datasets, colors, theme.theme, true);
```
Comment (vbersch) [inline packages/charts/src/components/DonutChart/index.tsx]: useChartData?

File: packages/charts/src/components/LineChart/index.tsx [hunk 3]
```diff
@@ -1,46 +1,42 @@
...
       getElementAtEvent,
       getDatasetAtEvent,
-      theme
-    } = this.props as LineChartPropTypes & ChartInternalProps;
+      width,
+      height,
+      noLegend
+    } = props;
 
     const chartOptions = mergeConfig(
```
Comment (vbersch) [inline packages/charts/src/components/LineChart/index.tsx]: This is no different than passing an object/function inline. useMemo?

File: packages/charts/src/components/PieChart/index.tsx [hunk 4]
```diff
@@ -1,38 +1,34 @@
...
-    } = this.props as PieChartPropTypes & ChartInternalProps;
+      options,
+      width,
+      height,
+      noLegend
+    } = props;
 
-    const doughnut = populateData(labels, datasets, colors, theme.theme, true);
+    const theme: any = useTheme();
+    const data = populateData(labels, datasets, colors, theme.theme, true);
```
Comment (vbersch) [inline packages/charts/src/components/PieChart/index.tsx]: useChartData?

File: packages/charts/src/components/RadarChart/index.tsx [hunk 5]
```diff
@@ -1,43 +1,33 @@
...
 
-    const bar = populateData(labels, datasets, colors, theme.theme);
-    bar.datasets.map(
-      (set) =>
-        (set.backgroundColor = HSLColor.of(set.backgroundColor)
-          .setAlpha(0.3)
-          .toString())
-    );
+    const theme: any = useTheme();
+    const data = populateData(labels, datasets, colors, theme.theme);
```
Comment (vbersch) [inline packages/charts/src/components/RadarChart/index.tsx]: useChartData?

File: packages/charts/src/components/RadialChart/index.tsx [hunk 6]
```diff
@@ -36,76 +36,75 @@ const styles = {
...
-      height: `${height}px`,
-      ...style
-    };
+  const classes = useStyles();
 
-    const outerClasses = StyleClassHelper.of(classes.radialChart);
-    if (className) {
-      outerClasses.put(className);
-    }
+  const radialChartContainerStyles = {
```
Comment (vbersch) [inline packages/charts/src/components/RadialChart/index.tsx]: useMemo?

File: packages/charts/src/internal/ChartContainer/withChartContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    const classes = useStyles(props);
+    let classNames = classes.chart;
+    if (className) {
+      classNames = `${classNames} ${className}`;
+    }
+
+    const loadingIndicator = getLoadingState(loading, datasets, (Component as any).LoadingPlaceholder);
+
+    const inlineStyle: CSSProperties = {
```
Comment (vbersch) [inline packages/charts/src/internal/ChartContainer/withChartContainer.tsx]: useMemo?

File: packages/charts/src/components/ColumnChart/index.tsx (hunk 1 above)
Comment (MarcusNotheis) [reply packages/charts/src/components/ColumnChart/index.tsx]: Done

File: packages/charts/src/components/PieChart/index.tsx (hunk 4 above)
Comment (MarcusNotheis) [reply packages/charts/src/components/PieChart/index.tsx]: done

File: packages/charts/src/components/RadarChart/index.tsx (hunk 5 above)
Comment (MarcusNotheis) [reply packages/charts/src/components/RadarChart/index.tsx]: done

File: packages/charts/src/components/RadialChart/index.tsx (hunk 6 above)
Comment (MarcusNotheis) [reply packages/charts/src/components/RadialChart/index.tsx]: Fixed

File: packages/charts/src/internal/ChartContainer/withChartContainer.tsx (hunk 7 above)
Comment (MarcusNotheis) [reply packages/charts/src/internal/ChartContainer/withChartContainer.tsx]: Fixed

**Pull request merged**